lib/src/hooks.dart

```diff
@@ -850,3 +850,119 @@ class _UseValueNotiferHookState<T>
     super.dispose();
   }
 }
+
+/// Creates an [ScrollController] automatically disposed.
+///
+/// [initialScrollOffset], [keepScrollOffset] and [debugLabel] are ignored after the first call.
+///
+/// See also:
+///   * [ScrollController]
+ScrollController useScrollController({
+  String debugLabel,
+  double initialScrollOffset = 0.0,
+  bool keepScrollOffset = true,
+  List keys,
+}) {
+  return Hook.use(_ScrollControllerHook(
+    debugLabel: debugLabel,
+    initialScrollOffset: initialScrollOffset,
+    keepScrollOffset: keepScrollOffset,
+    keys: keys,
+  ));
+}
+
+class _ScrollControllerHook extends Hook<ScrollController> {
+  final String debugLabel;
+  final double initialScrollOffset;
+  final bool keepScrollOffset;
+
+  const _ScrollControllerHook({
+    this.debugLabel,
+    this.initialScrollOffset,
+    this.keepScrollOffset,
+    List keys,
+  }) : super(keys: keys);
+
+  @override
+  _ScrollControllerHookState createState() => _ScrollControllerHookState();
+}
+
+class _ScrollControllerHookState
+    extends HookState<ScrollController, _ScrollControllerHook> {
+  ScrollController _scrollController;
+
+  @override
+  ScrollController build(BuildContext context) {
+    _scrollController ??= ScrollController(
```

Comment:
It's better to use `initHook` instead of `build` + `??=` to init the controller, and just leave the build method empty.